src/Core/Geographic/Coordinates.js

```diff
@@ -1,441 +1,250 @@
- * // y: 849862
- * // z: 23385912
- *
- * @return     {Position} - position
- */
+    set altitude(value) {
+        this.z = value;
+    }
 
-Coordinates.prototype.xyz = function xyz(target) {
-    _assertIsMetric(this.crs);
-    const v = target || new THREE.Vector3();
-    v.fromArray(this._values);
-    return v;
-};
+    get geodesicNormal() {
+        if (this._normalNeedsUpdate) {
+            this._normalNeedsUpdate = false;
 
-/**
- * Returns coordinates in the wanted [CRS]{@link http://inspire.ec.europa.eu/theme/rs}.
- * @example
- *
- * const position = { longitude: 2.33, latitude: 48.24, altitude: 24999549 };
- * const coords = new Coordinates('EPSG:4326', position.longitude, position.latitude, position.altitude); // Geographic system
- * const coordinates = coords.as('EPSG:4978'); // Geocentric system
- *
- * // or
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * const coordinates = coords.as('EPSG:4326');  // Geographic system
- *
- * //or
- *
- * new Coordinates('EPSG:4326', longitude: 2.33, latitude: 48.24, altitude: 24999549).as('EPSG:4978'); // Geocentric system
- *
- * // or
- *
- * new Coordinates('EPSG:4978', x: 20885167, y: 849862, z: 23385912).as('EPSG:4326'); // Geographic system
- *
- * @param      {string} - [crs]{@link http://inspire.ec.europa.eu/theme/rs} : Geocentric (ex: 'EPSG:4326') or Geographic (ex: 'EPSG:4978').
- * @return     {Position} - position
- */
+            if (CRS.is4326(this.crs)) {
+                ellipsoid.geodeticSurfaceNormalCartographic(this, this._normal);
+            } else if (this.crs == 'EPSG:4978') {
+                ellipsoid.geodeticSurfaceNormal(this, this._normal);
+            } else {
+                this._normal.set(0, 0, 1);
+            }
+        }
 
-Coordinates.prototype.as = function as(crs, target) {
-    if (crs === undefined || CRS.toUnit(crs) === undefined) {
-        throw new Error(`Invalid crs paramater value '${crs}'`);
+        return this._normal;
     }
-    return _convert(this, crs, target);
-};
 
-/**
- * Returns the normalized offset from top-left in extent of this Coordinates
- * e.g: extent.center().offsetInExtent(extent) would return (0.5, 0.5).
- * @param {Extent} extent
- * @param {Vector2} target optional Vector2 target. If not present a new one will be created
- * @return {Vector2} normalized offset in extent
- */
-Coordinates.prototype.offsetInExtent = function offsetInExtent(extent, target) {
-    if (this.crs != extent.crs) {
-        throw new Error('unsupported mix');
+    /**
+     * Return this Coordinates values into a `THREE.Vector3`.
+     *
+     * @param {THREE.Vector3} [target] - The target to put the values in. If not
+     * specified, a new vector will be created.
+     *
+     * @return {THREE.Vector3}
+     */
+    toVector3(target = new THREE.Vector3()) {
+        return target.set(this.x, this.y, this.z);
```

Comment:
you could write `return target.copy(this);`?

Reply:
Yes, well seen !